pallets/parachain-staking/src/lib.rs

```diff
@@ -380,11 +385,14 @@ pub mod pallet {
 			for x in &mut self.top_nominators {
 				if x.owner == nominator {
 					x.amount -= less;
-					self.total_counted -= less;
-					self.total_backing -= less;
+					// if there is at least 1 nominator in bottom nominators, compare it to check
+					// if it should be swapped with lowest top nomination and put in top
 					if let Some(top_bottom) = self.bottom_nominators.pop() {
 						if top_bottom.amount > x.amount {
 							new_top = Some(top_bottom);
+						} else {
+							// reset self.bottom_nominators
+							self.bottom_nominators.push(top_bottom);
```

Comment:
Without this line, the code above would pop the top bottom nomination every time this was called. This pop led to inconsistent `Collator2` state, preventing proper exit by unreserving reserved balance.

We can find the accounts that were incorrectly bumped by searching the `Collator2.nominators: Vec<AccountId>` for an account that is not in the top or bottom nominator sets. This is done in the migration code.

Reply:
This line is what patches the **bug** by the way. We forgot to push back after pop and it led to inadvertently removing nominations from `CollatorState`.